socs/agents/srs_cg635/agent.py

```diff
@@ -0,0 +1,237 @@
+                        data['data']['Running_State'] = self.clock.get_runs()
+                        data['data']['Timebase'] = self.clock.get_timebase()
+
+                        # get_lock_statuses returns a dict of the register bits
+                        # Loop through the items to add each to the data
+                        lock_statuses = self.clock.get_lock_statuses()
+                        for register, status in lock_statuses.items():
+                            # Not adding PLL causes a naming error
+                            # Two of the registers start with an number
+                            data['data']["PLL_" + register] = status
+
+                    except socket.timeout as e:
+                        self.log.error(f"Timeout in retrieving clock data: {e}")
+                        self.clock = None
+                        continue
+
+                    self.agent.publish_to_feed('srs_clock', data)
+
+                    # Allow this process to be queried to return current data
+                    session.data = data
+
+                else:
+                    self.log.warn("Could not acquire in monitor clock")
+
+            time.sleep(params['wait'])
+
+            if params['test_mode']:
+                break
+
+        return True, "Finished monitoring clock"
+
+    def _stop_acq(self, session, params):
+        """Stop monitoring the clock output."""
+        if self.monitor:
+            self.monitor = False
+            return True, 'requested to stop taking data.'
+        else:
+            return False, 'acq is not currently running'
+
+
+def make_parser(parser=None):
+    """Build the argument parser for the Agent. Allows sphinx to automatically
+    build documentation based on this function.
+
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    # Add options specific to this agent.
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--ip-address', type=str, help="Internal GPIB IP Address")
+    pgroup.add_argument('--gpib-slot', type=int, help="Internal SRS GPIB Address")
+    pgroup.add_argument('--mode', type=str, default='acq',
+                        choices=['idle', 'init', 'acq'],
+                        help="Starting action for the Agent.")
+
+    return parser
+
+
+def main(args=None):
+    # Start logging
+    txaio.start_logging(level=environ.get("LOGLEVEL", "info"))
+
+    parser = site_config.add_arguments()
+
+    # Get the default ocs agrument parser
+    parser = make_parser()
+
+    args = site_config.parse_args(agent_class='SRSCG635Agent',
+                                  parser=parser,
+                                  args=args)
+
+    if args.mode == 'init':
+        init_params = {'auto_acquire': False}
+    if args.mode == 'acq':
+        init_params = {'auto_acquire': True}
+
+    agent, runner = ocs_agent.init_site_agent(args)
+
+    p = SRSCG635Agent(agent, args.ip_address, int(args.gpib_slot))
```

Comment:
`args.gpib_slot` always be an int since its specified in the argparse so you can get rid of the int() casting here.